src/fsharp/symbols/Symbols.fs

```diff
@@ -75,12 +75,18 @@ module Impl =
          (fun (asmName, path) -> invalidOp (sprintf "The entity or value '%s' does not exist or is in an unresolved assembly. You may need to add a reference to assembly '%s'" path asmName))
          f
 
-    let makeReadOnlyCollection (arr: seq<'T>) = 
-        System.Collections.ObjectModel.ReadOnlyCollection<_>(Seq.toArray arr) :> IList<_>
-        
-    let makeXmlDoc (XmlDoc x) = makeReadOnlyCollection x
+    let makeReadOnlyCollection<'T> (items : 'T []) = 
+        System.Collections.ObjectModel.ReadOnlyCollection<'T>(items) :> IList<'T>
+
+    let makeReadOnlyCollectionOfList<'T> (items : 'T list) = 
+        System.Collections.ObjectModel.ReadOnlyCollection<'T>(Array.ofList items) :> IList<'T>
+
+    let emptyReadOnlyCollection<'T> = 
+        System.Collections.ObjectModel.ReadOnlyCollection<'T>([||]) :> IList<'T>
```

Comment:
`[||]` will also allocate here. Not sure, but maybe it'd be better to use `Array.Empty()` until the compiler can reuse empty arrays during code gen.

Reply:
makes sense